packages/builders/src/messages/poll/Poll.ts

```diff
@@ -0,0 +1,237 @@
+
+		this.data.answers.push(...resolved);
+		return this;
+	}
+
+	/**
+	 * Removes, replaces, or inserts answers for this poll.
+	 *
+	 * @remarks
+	 * This method behaves similarly
+	 * to {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice | Array.prototype.splice()}.
+	 * The maximum amount of answers that can be added is 10.
+	 *
+	 * It's useful for modifying and adjusting order of the already-existing answers of a poll.
+	 * @example
+	 * Remove the first answer:
+	 * ```ts
+	 * poll.spliceAnswers(0, 1);
+	 * ```
+	 * @example
+	 * Remove the first n answers:
+	 * ```ts
+	 * const n = 4;
+	 * poll.spliceAnswers(0, n);
+	 * ```
+	 * @example
+	 * Remove the last answer:
+	 * ```ts
+	 * poll.spliceAnswers(-1, 1);
+	 * ```
+	 * @param index - The index to start at
+	 * @param deleteCount - The number of answers to remove
+	 * @param answers - The replacing answer objects
+	 */
+	public spliceAnswers(
+		index: number,
+		deleteCount: number,
+		...answers: (APIPollMedia | PollAnswerBuilder | ((builder: PollAnswerBuilder) => PollAnswerBuilder))[]
+	): this {
+		const normalizedAnswers = normalizeArray(answers);
+		const resolved = normalizedAnswers.map((answer) => resolveBuilder(answer, PollAnswerBuilder));
+
+		this.data.answers.splice(index, deleteCount, ...resolved);
+		return this;
+	}
+
+	/**
+	 * Sets the answers for this poll.
+	 *
+	 * @remarks
+	 * This method is an alias for {@link PollBuilder.spliceAnswers}. More specifically,
+	 * it splices the entire array of answers, replacing them with the provided answers.
+	 *
+	 * You can set a maximum of 10 answers.
+	 * @param answers - The answers to set
+	 */
+	public setAnswers(
+		...answers: RestOrArray<APIPollMedia | PollAnswerBuilder | ((builder: PollAnswerBuilder) => PollAnswerBuilder)>
+	): this {
+		this.spliceAnswers(0, this.data.answers.length, ...normalizeArray(answers));
+		return this;
+	}
+
+	/**
+	 * Sets the question for this poll.
+	 *
+	 * @param options - The data to use for this poll's question
+	 */
+	public setQuestion(
+		options:
+			| Omit<APIPollMedia, 'emoji'>
+			| PollQuestionBuilder
+			| ((builder: PollQuestionBuilder) => PollQuestionBuilder),
+	): this {
+		this.data.question = resolveBuilder(options, PollQuestionBuilder);
+		return this;
+	}
+
+	/**
+	 * Updates the question of this poll.
+	 *
+	 * @param updater - The function to update the question with
+	 */
+	public updateQuestion(updater: (builder: PollQuestionBuilder) => void): this {
+		updater((this.data.question = new PollQuestionBuilder()));
+		return this;
+	}
+
+	/**
+	 * Sets the layout type for this poll.
+	 *
+	 * @remarks
+	 * This method is redundant while only one type of poll layout exists (`PollLayoutType.Default`)
+	 * with Discord using that as the layout type if none is specified.
+	 * @param type - The type of poll layout to use
+	 */
+	public setLayoutType(type = PollLayoutType.Default): this {
```

Comment:
I'm not convinced we want this default.. thoughts @vladfrangu?

Reply:
Discussed internally via https://canary.discord.com/channels/222078108977594368/999796393810067607/1292534311861489766